Update failed with org.apache.hc.core5.http.ConnectionClosedException

**Describe the bug**
When building a container image that pre-populates the dependency database from the `owasp/dependency-check` image:

```Dockerfile
FROM owasp/dependency-check:9.0.1

RUN --mount=type=secret,id=nvd,target=/kaniko/NVD_API_KEY \ 
    /usr/share/dependency-check/bin/dependency-check.sh --updateonly --nvdApiKey "$(cat /kaniko/NVD_API_KEY)"
```
 
Dependency Check fails to download the update with the following error:

```
[INFO] Checking for updates
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 6 time
[WARN] NVD API request failures are occurring; retrying request for the 7 time
[WARN] NVD API request failures are occurring; retrying request for the 8 time
[INFO] NVD API has 171,546 records in this update
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 6 time
[WARN] NVD API request failures are occurring; retrying request for the 6 time
[WARN] NVD API request failures are occurring; retrying request for the 6 time
[WARN] NVD API request failures are occurring; retrying request for the 6 time
[WARN] NVD API request failures are occurring; retrying request for the 7 time
[WARN] NVD API request failures are occurring; retrying request for the 7 time
[WARN] NVD API request failures are occurring; retrying request for the 7 time
[WARN] NVD API request failures are occurring; retrying request for the 7 time
[WARN] NVD API request failures are occurring; retrying request for the 8 time
[WARN] NVD API request failures are occurring; retrying request for the 8 time
[WARN] NVD API request failures are occurring; retrying request for the 8 time
[WARN] NVD API request failures are occurring; retrying request for the 8 time
[WARN] NVD API request failures are occurring; retrying request for the 9 time
[WARN] NVD API request failures are occurring; retrying request for the 9 time
[WARN] NVD API request failures are occurring; retrying request for the 9 time
[WARN] NVD API request failures are occurring; retrying request for the 9 time
[INFO] Downloaded 10,000/171,546 (6%)
[INFO] Recoverable I/O exception (org.apache.hc.core5.http.ConnectionClosedException) caught when processing request to {s}->https://services.nvd.nist.gov:443
[ERROR] Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@4c6f618f[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@6aca07d1[Wrapped task = org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$ScheduledRequestExecution@1bc553fc]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@520808f6[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 19]
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@4c6f618f[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@6aca07d1[Wrapped task = org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$ScheduledRequestExecution@1bc553fc]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@520808f6[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 19]
	at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2065)
	at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:833)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:340)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:562)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.execute(ScheduledThreadPoolExecutor.java:[70](https://gitlab.com/gitlab-ci-utils/docker-dependency-check/-/jobs/5619745796#L70)5)
	at java.base/java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:721)
	at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient.executeScheduled(InternalAbstractHttpAsyncClient.java:361)
	at org.apache.hc.client5.http.impl.async.AsyncHttpRequestRetryExec$1.failed(AsyncHttpRequestRetryExec.java:164)
	at org.apache.hc.client5.http.impl.async.AsyncProtocolExec$1.failed(AsyncProtocolExec.java:295)
	at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.failed(HttpAsyncMainClientExec.java:131)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.failed(ClientHttp1StreamHandler.java:285)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.disconnected(ClientHttp1StreamDuplexer.java:220)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onDisconnect(AbstractHttp1StreamDuplexer.java:409)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.disconnected(AbstractHttp1IOEventHandler.java:95)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.disconnected(ClientHttp1IOEventHandler.java:41)
	at org.apache.hc.core5.reactor.ssl.SSLIOSession$1.disconnected(SSLIOSession.java:247)
	at org.apache.hc.core5.reactor.InternalDataChannel.disconnected(InternalDataChannel.java:204)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.processClosedSessions(SingleCoreIOReactor.java:231)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.doTerminate(SingleCoreIOReactor.java:106)
	at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:93)
	at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
	at java.base/java.lang.Thread.run(Thread.java:840)
[INFO] Recoverable I/O exception (org.apache.hc.core5.http.ConnectionClosedException) caught when processing request to {s}->https://services.nvd.nist.gov:443
[ERROR] Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@3c27b846[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@61598479[Wrapped task = org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$ScheduledRequestExecution@67987bb4]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@484026a4[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 8]
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@3c27b846[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@61598479[Wrapped task = org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$ScheduledRequestExecution@67987bb4]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@484026a4[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 8]
	at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2065)
	at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:833)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:340)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:562)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.execute(ScheduledThreadPoolExecutor.java:705)
	at java.base/java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:721)
	at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient.executeScheduled(InternalAbstractHttpAsyncClient.java:361)
	at org.apache.hc.client5.http.impl.async.AsyncHttpRequestRetryExec$1.failed(AsyncHttpRequestRetryExec.java:164)
	at org.apache.hc.client5.http.impl.async.AsyncProtocolExec$1.failed(AsyncProtocolExec.java:295)
	at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.failed(HttpAsyncMainClientExec.java:131)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.failed(ClientHttp1StreamHandler.java:285)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.disconnected(ClientHttp1StreamDuplexer.java:220)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onDisconnect(AbstractHttp1StreamDuplexer.java:409)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.disconnected(AbstractHttp1IOEventHandler.java:95)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.disconnected(ClientHttp1IOEventHandler.java:41)
	at org.apache.hc.core5.reactor.ssl.SSLIOSession$1.disconnected(SSLIOSession.java:247)
	at org.apache.hc.core5.reactor.InternalDataChannel.disconnected(InternalDataChannel.java:204)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.processClosedSessions(SingleCoreIOReactor.java:231)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.doTerminate(SingleCoreIOReactor.java:106)
	at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:93)
	at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
	at java.base/java.lang.Thread.run(Thread.java:840)
[ERROR] Error updating the NVD Data
org.owasp.dependencycheck.data.update.exception.UpdateException: Error updating the NVD Data
	at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:338)
	at org.owasp.dependencycheck.data.update.NvdApiDataSource.update(NvdApiDataSource.java:110)
	at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:904)
	at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:876)
	at org.owasp.dependencycheck.App.runUpdateOnly(App.java:426)
	at org.owasp.dependencycheck.App.run(App.java:1[71](https://gitlab.com/gitlab-ci-utils/docker-dependency-check/-/jobs/5619745796#L71))
	at org.owasp.dependencycheck.App.main(App.java:88)
Caused by: io.github.jeremylong.openvulnerability.client.nvd.NvdApiException: NVD Returned Status Code: 504
	at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient.next(NvdCveClient.java:346)
	at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:317)
	... 6 common frames omitted
[INFO] Updating CISA Known Exploited Vulnerability list: https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json
[INFO] Begin database defrag
[INFO] End database defrag (604 ms)
[ERROR] Error updating the NVD Data
org.owasp.dependencycheck.data.update.exception.UpdateException: Error updating the NVD Data
	at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:338)
	at org.owasp.dependencycheck.data.update.NvdApiDataSource.update(NvdApiDataSource.java:110)
	at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:904)
	at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:8[76](https://gitlab.com/gitlab-ci-utils/docker-dependency-check/-/jobs/5619745796#L76))
	at org.owasp.dependencycheck.App.runUpdateOnly(App.java:426)
	at org.owasp.dependencycheck.App.run(App.java:171)
	at org.owasp.dependencycheck.App.main(App.java:88)
Caused by: io.github.jeremylong.openvulnerability.client.nvd.NvdApiException: NVD Returned Status Code: 504
	at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient.next(NvdCveClient.java:346)
	at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:317)
	... 6 common frames omitted
[INFO] Shutdown hook activated. Shutdown was not called. Shutting down JCS.
[INFO] Element event queue destroyed: org.apache.commons.jcs3.engine.control.event.ElementEventQueue@77d67cf3
...
ERROR: Job failed: exit code 8
```

**Version of dependency-check used**

Dependency Check CLI v9.0.1 (from the `owasp/dependency-check` image).

**To Reproduce**

1. Build the Dockerfile show above. In some cases it happens quickly (within 5 minutes), in some cases later (e.g. 30 minutes). It does not always happen, earlier today a couple of successful builds with 9.0.1 were completed (although tonight the NVD API seems more heavily loaded).

Hi, I'm using Version 9.0.1, i think we found the similar issue 

In my case i'm using Docker to `owasp/dependency-check:9.0.1` and we re-build it to update the database for using with our CI/CD, still got 504 when updating the NVD database.

`#7 4.528 [INFO] Instance is null, returning unconfigured instance                                                                                                                                                                                                                                                                                                                                                    
#7 4.913 [INFO] Using system property [[jcs.logSystem.mute] [true]]                                                                                                                                                                                                                                                                                                                                                  
#7 4.914 [INFO] Using system property [[jcs.logSystem] [slf4j]]                                                                                                                                                                                                                                                                                                                                                      
#7 4.930 [INFO] Setting default auxiliaries to "ODC"                                                                                                                                                                                                                                                                                                                                                                 
#7 4.994 [INFO] setting defaultCompositeCacheAttributes to [ useLateral = true, useRemote = true, useDisk = true, maxObjs = 0, maxSpoolPerRun = -1, diskUsagePattern = UPDATE, spoolChunkSize = 2 ]
#7 5.014 [INFO] setting defaultElementAttributes to [ IS_LATERAL = false, IS_SPOOL = true, IS_REMOTE = false, IS_ETERNAL = false, MaxLifeSeconds = 86400, IdleTime = 1800, CreateTime = 1701067853412, LastAccessTime = 1701067853412, getTimeToLiveSeconds() = 86399, createTime = 1701067853412 ]
#7 5.045 [INFO] initialized MemoryCache for CENTRAL
#7 5.045 [INFO] Constructed cache with name [CENTRAL] and cache attributes [ useLateral = true, useRemote = true, useDisk = true, maxObjs = 0, maxSpoolPerRun = -1, diskUsagePattern = UPDATE, spoolChunkSize = 2 ]
#7 5.085 [INFO] No cache event logger defined for auxiliary [jcs.auxiliary.ODC]
#7 5.096 [INFO] Using standard serializer [org.apache.commons.jcs3.utils.serialization.StandardSerializer@5b202a3a] for auxiliary [jcs.auxiliary.ODC]
#7 5.219 [INFO] Region [CENTRAL] : Set maxKeySize to: "1,000,000"
#7 5.224 [INFO] Region [CENTRAL] : Cache file root directory: /usr/share/dependency-check/data/cache
#7 5.228 [INFO] Region [CENTRAL] : Indexed Disk Cache is alive.
#7 5.241 [INFO] initialized MemoryCache for POM
#7 5.241 [INFO] Constructed cache with name [POM] and cache attributes [ useLateral = true, useRemote = true, useDisk = true, maxObjs = 0, maxSpoolPerRun = -1, diskUsagePattern = UPDATE, spoolChunkSize = 2 ]
#7 5.244 [INFO] No cache event logger defined for auxiliary [jcs.auxiliary.ODC]
#7 5.244 [INFO] Using standard serializer [org.apache.commons.jcs3.utils.serialization.StandardSerializer@333dd51e] for auxiliary [jcs.auxiliary.ODC]
#7 5.245 [INFO] Region [POM] : Set maxKeySize to: "1,000,000"
#7 5.245 [INFO] Region [POM] : Cache file root directory: /usr/share/dependency-check/data/cache
#7 5.246 [INFO] Region [POM] : Indexed Disk Cache is alive.
#7 5.249 [INFO] initialized MemoryCache for NODEAUDIT
#7 5.250 [INFO] Constructed cache with name [NODEAUDIT] and cache attributes [ useLateral = true, useRemote = true, useDisk = true, maxObjs = 0, maxSpoolPerRun = -1, diskUsagePattern = UPDATE, spoolChunkSize = 2 ]
#7 5.252 [INFO] No cache event logger defined for auxiliary [jcs.auxiliary.ODC]
#7 5.252 [INFO] Using standard serializer [org.apache.commons.jcs3.utils.serialization.StandardSerializer@52d645b1] for auxiliary [jcs.auxiliary.ODC]
#7 5.254 [INFO] Region [NODEAUDIT] : Set maxKeySize to: "1,000,000"
#7 5.254 [INFO] Region [NODEAUDIT] : Cache file root directory: /usr/share/dependency-check/data/cache
#7 5.256 [INFO] Region [NODEAUDIT] : Indexed Disk Cache is alive.
#7 5.257 [INFO] Parsed regions [CENTRAL, POM, NODEAUDIT]
#7 5.262 [INFO] Finished configuration in 334 ms.
#7 7.474 [INFO] Checking for updates
#7 102.2 [INFO] NVD API has 171,546 records in this update
#7 233.7 [INFO] Downloaded 10,000/171,546 (6%)
#7 482.3 [INFO] Downloaded 20,000/171,546 (12%)
#7 643.3 [INFO] Recoverable I/O exception (org.apache.hc.core5.http.ConnectionClosedException) caught when processing request to {s}->https://services.nvd.nist.gov:443
#7 643.3 [INFO] Recoverable I/O exception (org.apache.hc.core5.http.ConnectionClosedException) caught when processing request to {s}->https://services.nvd.nist.gov:443
#7 643.3 [INFO] Recoverable I/O exception (org.apache.hc.core5.http.ConnectionClosedException) caught when processing request to {s}->https://services.nvd.nist.gov:443
#7 800.8 [INFO] Downloaded 30,000/171,546 (17%)
#7 821.0 [WARN] NVD API request failures are occurring; retrying request for the 5 time
#7 824.5 [WARN] NVD API request failures are occurring; retrying request for the 5 time
#7 826.6 [ERROR] Error updating the NVD Data
#7 826.6 org.owasp.dependencycheck.data.update.exception.UpdateException: Error updating the NVD Data
#7 826.6        at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:338)
#7 826.6        at org.owasp.dependencycheck.data.update.NvdApiDataSource.update(NvdApiDataSource.java:110)
#7 826.6        at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:904)
#7 826.6        at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:876)
#7 826.6        at org.owasp.dependencycheck.App.runUpdateOnly(App.java:426)
#7 826.6        at org.owasp.dependencycheck.App.run(App.java:171)
#7 826.6        at org.owasp.dependencycheck.App.main(App.java:88)
#7 826.6 Caused by: io.github.jeremylong.openvulnerability.client.nvd.NvdApiException: NVD Returned Status Code: 504
#7 826.6        at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient.next(NvdCveClient.java:346)
#7 826.6        at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:317)
#7 826.6        ... 6 common frames omitted
#7 838.1 [ERROR] Error updating the NVD Data
#7 838.1 org.owasp.dependencycheck.data.update.exception.UpdateException: Error updating the NVD Data
#7 838.1        at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:338)
#7 838.1        at org.owasp.dependencycheck.data.update.NvdApiDataSource.update(NvdApiDataSource.java:110)
#7 838.1        at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:904)
#7 838.1        at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:876)
#7 838.1        at org.owasp.dependencycheck.App.runUpdateOnly(App.java:426)
#7 838.1        at org.owasp.dependencycheck.App.run(App.java:171)
#7 838.1        at org.owasp.dependencycheck.App.main(App.java:88)
#7 838.1 Caused by: io.github.jeremylong.openvulnerability.client.nvd.NvdApiException: NVD Returned Status Code: 504
#7 838.1        at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient.next(NvdCveClient.java:346)
#7 838.1        at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:317)
#7 838.1        ... 6 common frames omitted
#7 838.2 [INFO] Shutdown hook activated. Shutdown was not called. Shutting down JCS.
#7 838.2 [INFO] Element event queue destroyed: org.apache.commons.jcs3.engine.control.event.ElementEventQueue@77d67cf3
#7 838.2 [INFO] In DISPOSE, [NODEAUDIT] fromRemote [false]
#7 838.2 [INFO] In DISPOSE, [NODEAUDIT] auxiliary [NODEAUDIT]
#7 838.2 [INFO] In DISPOSE, [NODEAUDIT] put 0 into auxiliary [NODEAUDIT]
#7 838.2 [INFO] In dispose, destroying event queue.
#7 838.2 [INFO] Cache event queue destroyed: CacheEventQueue [listenerId=260771630, cacheName=NODEAUDIT]
#7 838.2 [INFO] Region [NODEAUDIT] : Saving keys to: NODEAUDIT, key count: 0
#7 838.2 [INFO] Region [NODEAUDIT] : Finished saving keys.
#7 838.2 [INFO] Region [NODEAUDIT] : Shutdown complete.
#7 838.2 [INFO] In DISPOSE, [NODEAUDIT] disposing of memory cache.
#7 838.2 [INFO] Memory Cache dispose called.
#7 838.2 [INFO] In DISPOSE, [CENTRAL] fromRemote [false]
#7 838.2 [INFO] In DISPOSE, [CENTRAL] auxiliary [CENTRAL]
#7 838.2 [INFO] In DISPOSE, [CENTRAL] put 0 into auxiliary [CENTRAL]
#7 838.2 [INFO] In dispose, destroying event queue.
#7 838.2 [INFO] Cache event queue destroyed: CacheEventQueue [listenerId=260771630, cacheName=CENTRAL]
#7 838.2 [INFO] Region [CENTRAL] : Saving keys to: CENTRAL, key count: 0
#7 838.2 [INFO] Region [CENTRAL] : Finished saving keys.
#7 838.2 [INFO] Region [CENTRAL] : Shutdown complete.
#7 838.2 [INFO] In DISPOSE, [CENTRAL] disposing of memory cache.
#7 838.2 [INFO] Memory Cache dispose called.
#7 838.2 [INFO] In DISPOSE, [POM] fromRemote [false]
#7 838.2 [INFO] In DISPOSE, [POM] auxiliary [POM]
#7 838.2 [INFO] In DISPOSE, [POM] put 0 into auxiliary [POM]
#7 838.2 [INFO] In dispose, destroying event queue.
#7 838.2 [INFO] Cache event queue destroyed: CacheEventQueue [listenerId=260771630, cacheName=POM]
#7 838.2 [INFO] Region [POM] : Saving keys to: POM, key count: 0
#7 838.2 [INFO] Region [POM] : Finished saving keys.
#7 838.2 [INFO] Region [POM] : Shutdown complete.
#7 838.2 [INFO] In DISPOSE, [POM] disposing of memory cache.
#7 838.2 [INFO] Memory Cache dispose called.
#7 838.2 [INFO] In dispose, destroying event queue.
#7 838.2 [ERROR] Region [NODEAUDIT] : Not alive and dispose was called, filename: NODEAUDIT
#7 838.2 [INFO] In dispose, destroying event queue.
#7 838.2 [ERROR] Region [CENTRAL] : Not alive and dispose was called, filename: CENTRAL
#7 838.2 [INFO] In dispose, destroying event queue.
#7 838.2 [ERROR] Region [POM] : Not alive and dispose was called, filename: POM`

**Describe the bug**
I downloaded the latest version 9.0.1 for the first time and encountered an error updating NVD data while scanning the project using nvdApiKey in command-line mode. i think we found the similar issue.

**Version of dependency-check used**
The problem occurs using version 9.0.1 of the cli.

**Log file**
[INFO] Checking for updates
[INFO] NVD API has 171,546 records in this update
[INFO] Downloaded 10,000/171,546 (6%)
[INFO] Downloaded 20,000/171,546 (12%)
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[INFO] Downloaded 30,000/171,546 (17%)
[INFO] Downloaded 40,000/171,546 (23%)
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[INFO] Downloaded 50,000/171,546 (29%)
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 6 time
[WARN] NVD API request failures are occurring; retrying request for the 7 time
[WARN] NVD API request failures are occurring; retrying request for the 6 time
[WARN] NVD API request failures are occurring; retrying request for the 8 time
[INFO] Downloaded 60,000/171,546 (35%)
[WARN] NVD API request failures are occurring; retrying request for the 9 time
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 6 time
[INFO] Downloaded 70,000/171,546 (41%)
[WARN] NVD API request failures are occurring; retrying request for the 7 time
[INFO] Recoverable I/O exception (org.apache.hc.core5.http.ConnectionClosedException) caught when processing request to {s}->https://services.nvd.nist.gov:443
[WARN] NVD API request failures are occurring; retrying request for the 5 time
[WARN] NVD API request failures are occurring; retrying request for the 6 time
[WARN] NVD API request failures are occurring; retrying request for the 7 time
[INFO] Recoverable I/O exception (org.apache.hc.core5.http.ConnectionClosedException) caught when processing request to {s}->https://services.nvd.nist.gov:443
[ERROR] Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@5cb78142[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@47be6433[Wrapped task = org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$ScheduledRequestExecution@1461f1f7]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@69f1ec5a[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 21]
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@5cb78142[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@47be6433[Wrapped task = org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$ScheduledRequestExecution@1461f1f7]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@69f1ec5a[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 21]
        at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2065)
        at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:833)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:340)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:562)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.execute(ScheduledThreadPoolExecutor.java:705)
        at java.base/java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:721)
        at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient.executeScheduled(InternalAbstractHttpAsyncClient.java:361)
        at org.apache.hc.client5.http.impl.async.AsyncHttpRequestRetryExec$1.failed(AsyncHttpRequestRetryExec.java:164)
        at org.apache.hc.client5.http.impl.async.AsyncProtocolExec$1.failed(AsyncProtocolExec.java:295)
        at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.failed(HttpAsyncMainClientExec.java:131)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.failed(ClientHttp1StreamHandler.java:285)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.disconnected(ClientHttp1StreamDuplexer.java:220)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onDisconnect(AbstractHttp1StreamDuplexer.java:409)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.disconnected(AbstractHttp1IOEventHandler.java:95)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.disconnected(ClientHttp1IOEventHandler.java:41)
        at org.apache.hc.core5.reactor.ssl.SSLIOSession$1.disconnected(SSLIOSession.java:247)
        at org.apache.hc.core5.reactor.InternalDataChannel.disconnected(InternalDataChannel.java:204)
        at org.apache.hc.core5.reactor.SingleCoreIOReactor.processClosedSessions(SingleCoreIOReactor.java:231)
        at org.apache.hc.core5.reactor.SingleCoreIOReactor.doTerminate(SingleCoreIOReactor.java:106)
        at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:93)
        at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
        at java.base/java.lang.Thread.run(Thread.java:833)
[INFO] Recoverable I/O exception (org.apache.hc.core5.http.ConnectionClosedException) caught when processing request to {s}->https://services.nvd.nist.gov:443
[ERROR] Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@3f4a9241[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@382f9a5c[Wrapped task = org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$ScheduledRequestExecution@5c77eaf2]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@2215d10a[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 10]
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@3f4a9241[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@382f9a5c[Wrapped task = org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$ScheduledRequestExecution@5c77eaf2]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@2215d10a[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 10]
        at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2065)
        at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:833)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:340)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:562)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.execute(ScheduledThreadPoolExecutor.java:705)
        at java.base/java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:721)
        at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient.executeScheduled(InternalAbstractHttpAsyncClient.java:361)
        at org.apache.hc.client5.http.impl.async.AsyncHttpRequestRetryExec$1.failed(AsyncHttpRequestRetryExec.java:164)
        at org.apache.hc.client5.http.impl.async.AsyncProtocolExec$1.failed(AsyncProtocolExec.java:295)
        at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.failed(HttpAsyncMainClientExec.java:131)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.failed(ClientHttp1StreamHandler.java:285)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.disconnected(ClientHttp1StreamDuplexer.java:220)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onDisconnect(AbstractHttp1StreamDuplexer.java:409)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.disconnected(AbstractHttp1IOEventHandler.java:95)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.disconnected(ClientHttp1IOEventHandler.java:41)
        at org.apache.hc.core5.reactor.ssl.SSLIOSession$1.disconnected(SSLIOSession.java:247)
        at org.apache.hc.core5.reactor.InternalDataChannel.disconnected(InternalDataChannel.java:204)
        at org.apache.hc.core5.reactor.SingleCoreIOReactor.processClosedSessions(SingleCoreIOReactor.java:231)
        at org.apache.hc.core5.reactor.SingleCoreIOReactor.doTerminate(SingleCoreIOReactor.java:106)
        at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:93)
        at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
        at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] Error updating the NVD Data
org.owasp.dependencycheck.data.update.exception.UpdateException: Error updating the NVD Data
        at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:338)
        at org.owasp.dependencycheck.data.update.NvdApiDataSource.update(NvdApiDataSource.java:110)
        at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:904)
        at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:709)
        at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:635)
        at org.owasp.dependencycheck.App.runScan(App.java:261)
        at org.owasp.dependencycheck.App.run(App.java:193)
        at org.owasp.dependencycheck.App.main(App.java:88)
Caused by: io.github.jeremylong.openvulnerability.client.nvd.NvdApiException: NVD Returned Status Code: 504
        at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient.next(NvdCveClient.java:346)
        at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:317)
        ... 7 common frames omitted
[INFO] Updating CISA Known Exploited Vulnerability list: https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json
[INFO] Begin database defrag
[INFO] End database defrag (3128 ms)
[WARN] Unable to update 1 or more Cached Web DataSource, using local data instead. Results may not include recent vulnerabilities.
[ERROR] Unable to continue dependency-check analysis.
[ERROR] One or more fatal errors occurred
[ERROR] Error updating the NVD Data
[ERROR] No documents exist
[INFO] Shutdown hook activated. Shutdown was not called. Shutting down JCS.
[INFO] Element event queue destroyed: org.apache.commons.jcs3.engine.control.event.ElementEventQueue@3fce8fd9
[INFO] In DISPOSE, [NODEAUDIT] fromRemote [false]
[INFO] In DISPOSE, [NODEAUDIT] auxiliary [NODEAUDIT]
[INFO] In DISPOSE, [NODEAUDIT] put 0 into auxiliary [NODEAUDIT]
[INFO] In dispose, destroying event queue.
[INFO] Cache event queue destroyed: CacheEventQueue [listenerId=259922143, cacheName=NODEAUDIT]
[INFO] Region [NODEAUDIT] : Saving keys to: NODEAUDIT, key count: 0
[INFO] Region [NODEAUDIT] : Finished saving keys.
[INFO] Region [NODEAUDIT] : Shutdown complete.
[INFO] In DISPOSE, [NODEAUDIT] disposing of memory cache.
[INFO] Memory Cache dispose called.
[INFO] In DISPOSE, [CENTRAL] fromRemote [false]
[INFO] In DISPOSE, [CENTRAL] auxiliary [CENTRAL]
[INFO] In DISPOSE, [CENTRAL] put 0 into auxiliary [CENTRAL]
[INFO] In dispose, destroying event queue.
[INFO] Cache event queue destroyed: CacheEventQueue [listenerId=259922143, cacheName=CENTRAL]
[INFO] Region [CENTRAL] : Saving keys to: CENTRAL, key count: 0
[INFO] Region [CENTRAL] : Finished saving keys.
[INFO] Region [CENTRAL] : Shutdown complete.
[INFO] In DISPOSE, [CENTRAL] disposing of memory cache.
[INFO] Memory Cache dispose called.
[INFO] In DISPOSE, [POM] fromRemote [false]
[INFO] In DISPOSE, [POM] auxiliary [POM]
[INFO] In DISPOSE, [POM] put 0 into auxiliary [POM]
[INFO] In dispose, destroying event queue.
[INFO] Cache event queue destroyed: CacheEventQueue [listenerId=259922143, cacheName=POM]
[INFO] Region [POM] : Saving keys to: POM, key count: 0
[INFO] Region [POM] : Finished saving keys.
[INFO] Region [POM] : Shutdown complete.
[INFO] In DISPOSE, [POM] disposing of memory cache.
[INFO] Memory Cache dispose called.
[INFO] In dispose, destroying event queue.
[ERROR] Region [NODEAUDIT] : Not alive and dispose was called, filename: NODEAUDIT
[INFO] In dispose, destroying event queue.
[ERROR] Region [CENTRAL] : Not alive and dispose was called, filename: CENTRAL
[INFO] In dispose, destroying event queue.
[ERROR] Region [POM] : Not alive and dispose was called, filename: POM

**To Reproduce**
Steps to reproduce the behavior:
1. dependency-check.sh --purge
2. ./dependency-check.sh -project "test" -scan /Test --nvdApiKey XXXXXX

--- a/pom.xml
+++ b/pom.xml
@@ -972,7 +972,7 @@ Copyright (c) 2012 - Jeremy Long
             <dependency>
                 <groupId>io.github.jeremylong</groupId>
                 <artifactId>open-vulnerability-clients</artifactId>
-                <version>5.0.2</version>
+                <version>5.0.3</version>
             </dependency>
             <dependency>
                 <groupId>org.anarres.jdiagnostics</groupId>

